### ISSUE TITLE
How to backup secondary roms ?

### ISSUE THREAD

**Issue by wazeeahmed** [chenxiaolong/DualBootPatcher#206]
@chenxiaolong Recently rr 5.7.4 got released . Previous version 5.7.3 is my current primary rom . I have to clean flash in order to get a bugfree experience . Can you tell me how to create backups for secondary roms ? At least data backups will suffice .

**Comment by Caio99BR** [comment]
It will be a feature of next versions of DualBootPatcher, it isn't official
available, but you can in the XDA thread for it :)

Em sáb, 24 de set de 2016 00:50, DarkHeart Z <notifications@github.com>
escreveu:

**Comment by chenxiaolong** [comment]
It is possible to do from the command line via `adb`. You'll need to extract the `assets/data-VERSION.tar.xz` file from the apk and then extract the `mbtool_recovery` file from the `tar.xz` file.

Then, reboot to recovery and run:

```sh
adb push mbtool_recovery /tmp/
adb shell mount /system
```

To back up a ROM, run:

```sh
adb shell /tmp/mbtool_recovery backup -r extsd-slot-cm -n my_backup_name -d /microSD/backups
```

and to restore a ROM:

```sh
adb shell /tmp/mbtool_recovery restore -r extsd-slot-cm -n my_backup_name -d /microSD/backups
```

(You can restore to a different slot if you want!)

**Comment by wazeeahmed** [comment]
@Caio99BR I did . But it only works on galaxy s4 . And nonetheless it has gui bugs which took me more than minutes get pass of . rlorange haven't worked on it in years 😐

**Comment by wazeeahmed** [comment]
@chenxiaolong Bro you could've implement it in bootui and app also . Will it come live in 9.0.0 ?

**Comment by chenxiaolong** [comment]
This is now implemented in 9.0.0 :) https://snapshots.noobdev.io/

**Comment by wazeeahmed** [comment]
You know what :) I'm gonna give ya a thumbs up which I already did and  give ya a 🙏
